pkg/kubelet/kubelet.go

```diff
@@ -2277,29 +2277,66 @@ func GetPhase(spec *api.PodSpec, info []api.ContainerStatus) api.PodPhase {
 	}
 }
 
+func readyPodCondition(isPodReady bool, reason, message string, existingStatus *api.PodStatus) []api.PodCondition {
+	condition := api.PodCondition{
+		Type: api.PodReady,
+	}
+	if isPodReady {
+		condition.Status = api.ConditionTrue
+	} else {
+		condition.Status = api.ConditionFalse
+	}
+	condition.Reason = reason
+	condition.Message = message
+	if existingStatus != nil {
+		if api.IsPodReadyConditionTrue(*existingStatus) != isPodReady {
+			// condition has transitioned, update transition time.
+			condition.LastTransitionTime = util.Now()
+		} else {
+			// retain the existing transition time.
+			existingCondition := api.GetPodReadyCondition(*existingStatus)
+			if existingCondition != nil {
+				condition.LastTransitionTime = existingCondition.LastTransitionTime
+			}
+		}
+	}
+	return []api.PodCondition{condition}
+}
+
 // getPodReadyCondition returns ready condition if all containers in a pod are ready, else it returns an unready condition.
-func getPodReadyCondition(spec *api.PodSpec, statuses []api.ContainerStatus) []api.PodCondition {
-	ready := []api.PodCondition{{
-		Type:   api.PodReady,
-		Status: api.ConditionTrue,
-	}}
-	unready := []api.PodCondition{{
-		Type:   api.PodReady,
-		Status: api.ConditionFalse,
-	}}
-	if statuses == nil {
-		return unready
+func getPodReadyCondition(spec *api.PodSpec, containerStatuses []api.ContainerStatus, existingStatus *api.PodStatus) []api.PodCondition {
+	// Find if all containers are ready or not.
+	if containerStatuses == nil {
+		return readyPodCondition(false, "Could not get container statuses", "", existingStatus)
```

Comment:
Shouldn't reason be a short and brief? "Could not get container statuses" should be message, not reason.